### PR TITLE
Fix shape label object obb verticies

### LINKED ISSUE #1128: SimpleITK LabelShapeStatisticsFilter GetOrientedBoundingBox Coordinates Appear Incorrect
@blowekamp asked me to submit this issue. 

Vertices produced by sitk.LabelShapeStatisticsFilter()'s GetOrientedBoundingBoxVertices() are properly in real-world coordinates of the label image only after applying a rotation.

Here is a minimal example that reproduces and solves the issue: https://gist.github.com/kevinalexbrown/d2474bc8245f9aa36af64fcc7c52e8f1

### PR DESCRIPTION
**Pull request by blowekamp**
closes #1128

## PR Checklist

- [X] :no_entry_sign: [Makes breaking changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes)
- [X] :no_entry_sign: [Makes design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes)
- [X] :no_entry_sign: Adds the License notice to new files.
- [X] :no_entry_sign: Adds Python wrapping.
- [X] :no_entry_sign: Adds tests and baseline comparison (quantitative).
- [X] :no_entry_sign: [Adds test data](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/UploadBinaryData.md).
- [X] :no_entry_sign: Adds Examples to [ITKExamples](https://github.com/InsightSoftwareConsortium/ITKExamples)
- [X] :no_entry_sign: Adds Documentation.

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by blowekamp** [comment]
@dzenanz @hjmjohnson CI came back clean please review.

File: Modules/Filtering/LabelMap/include/itkShapeLabelObject.h [hunk 1]
```diff
@@ -593,27 +593,21 @@ class ShapeLabelObject : public LabelObject<TLabel, VImageDimension>
...
-  /** Get an array of point verities which define the corners of the
-   * oriented bounding box.
+  /** Get an array of point vertices which define the corners of the
+   * oriented bounding box in physical space.
    *
-   * The first element in the array contains the minimum coordination
-   * values while the last contains the maximum. Use the index of the
-   * array in binary to determine min/max for the indexed vertex.
-   * For example, in 2D, binary  counting will give[0,0], [0,1],
+   * The first element in the array contains minimum coordination values
```
Comment (dzenanz) [inline Modules/Filtering/LabelMap/include/itkShapeLabelObject.h]: coordination -> coordinate

File: Modules/Filtering/LabelMap/test/itkShapeLabelMapFilterGTest.cxx [hunk 2]
```diff
@@ -429,6 +444,46 @@ TEST_F(ShapeLabelMapFixture, 2D_T1_1_FlipDirection)
...
+  image->SetPixel(MakeIndex(5, 8), 1);
+
+  DirectionType direction;
+
+  const double d[4] = { 0, 1.0, 1.0, 0 };
+
+  direction = DirectionType::InternalMatrixType(d);
+
+  image->SetDirection(direction);
+
```
Comment (dzenanz) [inline Modules/Filtering/LabelMap/test/itkShapeLabelMapFilterGTest.cxx]: Too much white-space for my taste. An empty line between each statement does not really make them stand out more.

Comment (blowekamp) [reply Modules/Filtering/LabelMap/test/itkShapeLabelMapFilterGTest.cxx]: This is compliant with our clang formatting configuration and ITK coding standards.

Comment (hjmjohnson) [reply Modules/Filtering/LabelMap/test/itkShapeLabelMapFilterGTest.cxx]: It is compliant, but so would @dzenanz suggestion.  You could also put 0 lines, or 2 lines between each statement, and that would also be compliant.  3 lines would not be compliant.

**Comment by blowekamp** [comment]
> nit-pick ... The test should have been the first patch, demonstrating failure, and then the fix after that.

Yes, I use to suggest that with Gerrit where the CI would run on each commit in the change set so the CI would demonstrate both the failure and the fix. With the current CI, it only runs the on the whole PR so it no longer can "demonstrates" the failure and the fix. I still like to have the fix and the test update separate so that if the patch needs to be back ported it is a little easer.

**Comment by hjmjohnson** [comment]
:)..  I happen to be lecturing on Red/Green refactoring today.

https://dzone.com/articles/pattern-of-the-month-red-green-refactor

**Comment by blowekamp** [comment]
The failing test on ITK.MacOS.Python: `itkPyBufferMemoryLeakTest` is not related.

**Pull request merged**